src/Database/AbstractModel.php

```diff
@@ -89,19 +99,11 @@ public function __construct(array $attributes = [])
      */
     public function getDates()
     {
-        static $dates = [];
-
-        $class = get_class($this);
-
-        if (! isset($dates[$class])) {
-            static::$dispatcher->dispatch(
-                new ConfigureModelDates($this, $this->dates)
-            );
-
-            $dates[$class] = $this->dates;
-        }
+        static::$dispatcher->dispatch(
+            new ConfigureModelDates($this, $this->dates)
+        );
 
-        return $dates[$class];
+        return array_merge($this->dates, Arr::get(static::$dateAttributes, static::class, []));
```

Comment:
Should we keep the cache semantics of the previous implementation - so that the event dispatch and the array merge only happens once per class per request?

(Not sure how often this is called internally... it might not be relevant any longer when the event dispatching is gone.)

Reply:
I'll take care of this!

Reply:
Played around with it. Static caching would cause problems in tests (uh oh), and I'm not too worried about it once the event dispatching is gone, so I'm leaving this as is.